Presentation.Web/app/components/data-processing/data-processing-subnav.controller.ts

```diff
@@ -1,22 +1,76 @@
-    app.config(["$stateProvider", $stateProvider => {
-        $stateProvider.state("data-processing", {
-            url: "/data-processing",
-            abstract: true,
-            template: "<ui-view autoscroll=\"false\" />"
-            ,
-            controller: ["$rootScope", "$scope", ($rootScope, $scope) => {
-                $rootScope.page.title = "Databehandling";
-                $rootScope.page.subnav = [
-                    { state: "data-processing.overview", text: "Databehandleraftaler" }
-                ];
-
-                $rootScope.subnavPositionCenter = false;
-
-                $scope.$on("$viewContentLoaded", () => {
-                    $rootScope.positionSubnav();
-                });
-            }]
-        });
-    }]);
-})(angular, app);
+﻿module Kitos.DataProcessing.Agreement.Edit {
+    "use strict";
+
+    export class SubNavDataProcessingAgreementController {
+        static $inject: Array<string> = [
+            "$scope",
+            "$rootScope",
+            "$state",
+            "$http",
+            "notify",
+            "dataProcessingAgreementService"
+        ];
+
+        constructor(
+            private $scope,
+            private $rootScope,
+            private $state: angular.ui.IStateService,
+            private $http,
+            private notify,
+            private dataProcessingAgreementService: Services.DataProcessing.IDataProcessingAgreementService) {
+
+            this.$scope.page.title = "Databehandleraftaler";
+ 
+            $rootScope.page.subnav = [
+                { state: "data-processing.overview", text: "Databehandleraftaler"}
+            ];
+            $rootScope.page.subnav.buttons = [
+                { func: remove, text: "Slet Databehandleraftale", style: "btn-danger", showWhen: "data-processing.edit-agreement", dataElementType: 'removeDataProcessingAgreementButton' }
+            ];
+
+            $rootScope.subnavPositionCenter = false;
+
+            $scope.$on("$viewContentLoaded", () => {
+                $rootScope.positionSubnav();
+            });
+
+            function remove() {
+                if (!confirm("Er du sikker på du vil slette Databehandleraftale?")) {
+                    return;
+                }
+                var dataProcessingAgreementId = $state.params["id"];
+                var msg = notify.addInfoMessage("Sletter Databehandleraftale...", false);
+
+                dataProcessingAgreementService.delete(dataProcessingAgreementId).then(
+                    deleteResponse => {
+                        msg.toSuccessMessage("Databehandleraftale slettet!");
+                            $state.go("data-processing.overview");
+                        },
+                    (errorResponse: Models.Api.ApiResponseErrorCategory) => {
+                        switch (errorResponse) {
+                        case Models.Api.ApiResponseErrorCategory.BadInput:
+                        case Models.Api.ApiResponseErrorCategory.Conflict:
```

Comment:
Conflict skal ikke være en case her. Det kan "not found" til gengæld så udvid "mapperen" og "kategorien" til at indeholde NotFound og så håndter den her på forskellig vis dvs. notfound = besked1 og alt andet kan du faktisk ikke svare andet på end det der står i din "default"